indico/modules/users/controllers.py

```diff
@@ -1002,3 +1005,36 @@ def _process_DELETE(self):
         logger.info('User %s unblocked %s', session.user, self.user)
         flash(_('{name} has been unblocked.').format(name=self.user.name), 'success')
         return jsonify(success=True)
+
+
+class RHUserDelete(RHUserBase):
+    """Delete a user.
+
+    Deletes the user, and all their associated data. If it is not possible to delete the user, it will
+    instead fallback to anonymizing the user.
+    """
+
+    def _check_access(self):
+        RHUserBase._check_access(self)
+        if not session.user.is_admin or not config.ENABLE_DELETE_USER_FROM_UI:
+            raise Forbidden
+        if self.user == session.user:
+            raise Forbidden(_('You cannot delete your own account'))
+
+    def _process(self):
+        name = self.user.name
+        signals.users.db_deleted.send(self.user, flushed=False)
+        try:
+            db.session.delete(self.user)
+            db.session.flush()
+        except IntegrityError as exc:
+            db.session.rollback()
+            logger.info('User %s could not be deleted %s', name, str(exc))
+            anonymize_user(self.user)
+            logger.info('User %s anonymized %s', session.user, name)
```

Comment:
```suggestion
            logger.info('User %r anonymized %s', session.user, user_repr)
```

You need to set `user_repr = repr(user)` before starting to delete the user, since down here it might no longer work.